lib/checkfunctions.cpp

```diff
@@ -682,3 +682,108 @@ void CheckFunctions::copyElisionError(const Token *tok)
                 "Using std::move for returning object by-value from function will affect copy elision optimization."
                 " More: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-return-move-local");
 }
+
+void CheckFunctions::useStandardLibrary()
+{
+    if (!mSettings->severity.isEnabled(Severity::style))
+        return;
+
+    for (const Scope& scope: mTokenizer->getSymbolDatabase()->scopeList) {
+        if (scope.type != Scope::ScopeType::eFor)
+            continue;
+
+        const auto *tok = scope.bodyStart; // { token
+        const auto *forToken = tok->previous()->link()->previous(); // should be for
+
+        // for ( firstExpression ; secondExpression ; thirdExpression )
+        const Token *firstExpression  = forToken->next()->astOperand2()->astOperand1();
+        if (!firstExpression)
+            continue;
+        const auto *astOperand22 = forToken->next()->astOperand2()->astOperand2();
+        if (!astOperand22)   // #7050
+            continue;
+        const Token *secondExpression = astOperand22->astOperand1();
+        if (!secondExpression)
+            continue;
+        const Token *thirdExpression = astOperand22->astOperand2();
+        if (!thirdExpression)
+            continue;
+
+        // 1. we expect that idx variable will be initialized with 0
+        const auto* idxToken = firstExpression->astOperand1();
+        const auto* initVal = firstExpression->astOperand2();
+        if (!idxToken || !initVal || !initVal->isNumber() || initVal->str() != "0")
+            continue;
+        const auto idxVarId = idxToken->varId();
+
+        // 2. we expect that idx will be less of something
+        if (!secondExpression->isComparisonOp())
+            continue;
+        const auto& secondOp = secondExpression->str();
+
+        if (!(("<" == secondOp && secondExpression->astOperand1()->varId() == idxVarId) ||
+              (">" == secondOp && secondExpression->astOperand2()->varId() == idxVarId)))
+            continue;
+
+        // 3. we expect idx incrementing by 1
+        const bool inc = thirdExpression->str() == "++" && thirdExpression->astOperand1()->varId() == idxVarId;
+        const bool plusOne = thirdExpression->str() == "+=" &&
+                             thirdExpression->astOperand1()->varId() == idxVarId &&
+                             thirdExpression->astOperand2() != nullptr && thirdExpression->astOperand2()->str() == "1";
+        if (!inc && !plusOne)
+            continue;
+
+        // technically using void* here is not correct but some compilers could allow it
+
+        const static std::string memcpyName = mTokenizer->isCPP() ? "std::memcpy" : "memcpy";
```

Comment:
I would rather not have a static variable here. Different files in a project may be C and C++.

Reply:
Thank you, it's a good point